tools/db/build/build_models_script.php

```diff
@@ -0,0 +1,330 @@
+      
+      $this->unzip() || fail("Unable to extract model_info.zip");
+      
+      if(($v = $this->build()) === false) 
+        fail("Unable to build models.sql");
+      file_put_contents($this->cache_path . "models.sql", $v) || fail("Unable to write models.sql to " . $this->cache_path);
+      
+      return true;
+    }
+
+    /**
+      Build a SQL script to insert model_info data into the database
+    */
+
+    private $models = array();
+    private $link;
+    
+    function unzip() {
+      $zip = new ZipArchive();
+      if(!$zip->open($this->cache_path . 'model_info.zip', 0)) 
+        return false;
+      
+      $result = $zip->extractTo($this->models_path);
+      $result |= $zip->close();
+      
+      return $result;
+    }
+
+    /**
+      Search through all folders under models_path to find .model_info files to
+      import into the database
+    */
+    function build() {
+      $this->link = new mysqli();
+
+      if(empty($this->models_path)) {
+        return false;
+      }
+      
+      $files = glob($this->models_path . '*.model_info');
+      foreach($files as $file) {
+        if(!$this->process_file($file)) {
+          return false;
+        }
+      }
+      
+      return 
+        $this->generate_model_inserts() .
+        $this->generate_model_language_inserts() .
+        $this->generate_model_link_inserts() .
+        $this->generate_model_related_inserts();
+    }
+    
+    /**
+      Loads a .model_info file, parses it into the 
+      models array
+    */
+    function process_file($file) {
+      if(($data = file_get_contents($file)) === false) {
+        return false;
+      }
+      
+      $model = json_decode($data);
+      $model->json = $data;
+      
+      array_push($this->models, $model);     
+      return true;
+    }
+    
+    /**
+      Generate an SQL script to insert entries in to the t_model table
+    */
+    function generate_model_inserts() {
+      $result = <<<END
+        INSERT t_model (
+          model_id, 
+          name, 
+          author_name, 
+          author_email, 
+          description, 
+          license, 
+          last_modified, 
+          version,
+
+          min_keyman_version,
+          min_keyman_version_1,
+          min_keyman_version_2,
+          
+          package_filename,
+          package_filesize,
+          js_filename,
+          js_filesize,
+
+          is_rtl,
+          
+          includes_fonts,
+          
+          model_info
+        ) VALUES
+END;
+
+      $comma = '';
+      foreach($this->models as $model) {
+        $isRTL = isset($model->isRTL) && $model->isRTL;
+        
+        if(isset($model->minKeymanVersion) && preg_match('/^(\d+)\.(\d+)/', $model->minKeymanVersion, $matches)) {
+          $minKeymanVersion1 = $matches[1];
+          $minKeymanVersion2 = $matches[2];
+        } else {
+          $minKeymanVersion1 = 9;
```

Comment:
Isn't `$minKeymanVersion1` going to be 12?

Reply:
Yes, but it shouldn't matter, because minKeymanVersion will always be set in the .model_info files by the compiler. This is silly safety checks. Will fix.